pac4j-saml-db/src/main/java/org/pac4j/saml/dbmetadata/SAML2IdentityProviderMetadataResolver2.java

```diff
@@ -0,0 +1,144 @@
+package org.pac4j.saml.dbmetadata;
+
+import java.io.StringReader;
+import java.util.Iterator;
+
+import javax.annotation.Nullable;
+
+import org.opensaml.core.criterion.EntityIdCriterion;
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.saml.metadata.resolver.MetadataResolver;
+import org.opensaml.saml.metadata.resolver.impl.DOMMetadataResolver;
+import org.opensaml.saml.saml2.metadata.EntitiesDescriptor;
+import org.opensaml.saml.saml2.metadata.EntityDescriptor;
+import org.opensaml.xml.util.XMLHelper;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.saml.exceptions.SAMLException;
+import org.pac4j.saml.metadata.SAML2IdentityProviderMetadataResolver;
+import org.pac4j.saml.metadata.SAML2MetadataResolver;
+import org.pac4j.saml.util.Configuration;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import net.shibboleth.utilities.java.support.component.ComponentInitializationException;
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+import net.shibboleth.utilities.java.support.xml.XMLParserException;
+
+
+/**
+ * Alternative to {@link SAML2IdentityProviderMetadataResolver} that doesn't take a path to a metadata file but directly metadata content.
+ *
+ * TODO: We cannot extend {@link SAML2IdentityProviderMetadataResolver} because it needs a path in the constructor. It would be good to have
+ * a common ancestor to reuse some code.
+ * 
+ * @author jkacer
+ */
+public class SAML2IdentityProviderMetadataResolver2 implements SAML2MetadataResolver {

```

Comment:
Done. I added an abstract class (AbstractSAML2IdentityProviderMetadataResolver) and the two implementations now extend it, overriding/implementing only a few different methods. Impls: PathBasedSAML2IdentityProviderMetadataResolver and StringBasedSAML2IdentityProviderMetadataResolver.

Sorry for slow progress, I have had very little time to work on PAC4J recently.

I'll continue with those changes in Clients (discussion below) and with merging changes from PAC4J/master, I see some code conflicts now.